### ISSUE TITLE
Add Parse/TryParse methods

### ISSUE THREAD

**Issue by xaviergonz** [adamreeve/semver.net#10]
just for consistency with for example ints

**Comment by T3nak** [comment]
https://github.com/maxhauser/semver seems to expose [Parse](https://github.com/maxhauser/semver/blob/master/src/Semver/SemVersion.cs#L108) and [TryParse](https://github.com/maxhauser/semver/blob/master/src/Semver/SemVersion.cs#L165) methods, which is preferable in the majority of cases.

1. A simple Boolean check is much cleaner than catching an exception. Consider:
```c#
if (someCondition && SemVer.Version.TryParse(version, out SemVer.Version semVer))
{
    // Happy coding
}
```
vs.:
```c#
if (someCondition)
{
    try
    {
        var semVer = new SemVer.Version(version);
    }
    catch (ArgumentException)
    {
        // Handle parsing failure
    }
}
```
2. When you are in a context where an ArgumentException can be thrown for several reasons - how would you know if it was actually thrown because semver parsing failed? Using try-catch blocks for performing logic seems unnecessary.
3. [C# 7 out values inline as arguments](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7#out-variables) (ties to point 1) keeps code succinct and tidy.

If you are willing to accept, I will be humbled to submit a PR for you to review. You clearly have spent lots of effort and time on this library. It is pleasing to know that this library is still receiving attention from its author since 2016.

**Comment by adamreeve** [comment]
Hi @T3nak, yes I'd be happy to accept a PR for this feature if you'd like to implement it, I don't have much time myself to work on feature but am happy to accept appropriate PRs.

**Comment by adamreeve** [comment]
This was added in #48 and released in version 1.3.0
